src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -5,95 +5,136 @@ Test coverage:
+  - Also the instance step mode vertex buffer OOB are CPU validated for drawIndexed, so we only test
+    robustness access for vertex step mode vertex buffers.
+
+2) Draw call parameter (vertexCount, firstVertex, indexCount, firstIndex, baseVertex, instanceCount,
+  firstInstance, vertexCountInIndexBuffer)
+  - The parameter which will go out of bounds. Filtered depending on the draw call type.
+  - vertexCount, firstVertex: used for drawIndirect only, test for vertex step mode buffer OOB
+  - instanceCount, firstInstance: used for both drawIndirect and drawIndexedIndirect, test for
+    instance step mode buffer OOB
+  - baseVertex, vertexCountInIndexBuffer: used for both drawIndexed and drawIndexedIndirect, test
+    for vertex step mode buffer OOB. vertexCountInIndexBuffer indicates how many vertices are used
+    within the index buffer, i.e. [0, 1, ..., vertexCountInIndexBuffer-1].
+  - indexCount, firstIndex: used for drawIndexedIndirect only, validate the vertex buffer access
+    when the vertex itself is OOB in index buffer. This won't happen in drawIndexed as we have index
+    buffer OOB CPU validation for it.
+
+3) Attribute type (float32, float32x2, float32x3, float32x4)
   - The input attribute type in the vertex shader
 
-5) Error scale (1, 4, 10^2, 10^4, 10^6)
+4) Error scale (0, 1, 4, 10^2, 10^4, 10^6)
   - Offset to add to the correct draw call parameter
+  - 0 For control case
 
-6) Additional vertex buffers (0, +4)
+5) Additional vertex buffers (0, +4)
   - Tests that no OOB occurs if more vertex buffers are used
 
-The tests will also have another vertex buffer bound for an instanced attribute, to make sure
-instanceCount / firstInstance are tested.
+6) Partial last number and offset vertex buffer (false, true)
+  - Tricky cases that make vertex buffer OOB.
+  - Partial last number will make vertex buffer 1 byte less than enough, making the last vertex OOB
+    with 1 byte.
+  - Offset vertex buffer will bind the vertex buffer to render pass with 4 bytes offset, causing OOB
+  - For drawIndexed, these two flag will be surpressed for instance step mode vertex buffer to make
+    sure it pass the CPU validation.
+
+The tests will have one instance step mode vertex buffer bound for instanced attributes, to make
+sure instanceCount / firstInstance are tested.
 
 The tests will include multiple attributes per vertex buffer.
 
-The vertex buffers will be filled by repeating a few chosen values until the end of the buffer.
+The vertex buffers will be filled by repeating a few values randomly chosen for each test until the
+end of the buffer.
 
 The test will run a render pipeline which verifies the following:
-1) All vertex attribute values occur in the buffer or are zero
+1) All vertex attribute values occur in the buffer or are 0 (for control case it can't be 0)
 2) All gl_VertexIndex values are within the index buffer or 0
 
 TODO:
```

Comment:
Now I hit the situation that on Windows / D3D12 the `VertexIndex` count from 0 rather than `baseVertex`, causing the `indexInBound` validation, i.e. 
```
var indexInBounds : bool = VertexIndex == 0u ||
      (VertexIndex >= ${vertexIndexOffset}u &&
      VertexIndex < ${vertexIndexOffset + numVertices}u)
```
failed. I am changing it to fit for both situation, i.e. count from 0 and count from baseVertex.

Reply:
However I think in spec https://www.w3.org/TR/webgpu/#abstract-opdef-resolve-indices, during Index Resolution, the baseVertex should be added into vertexIndex. Is this a implementation bug?

Reply:
Yes it is an implementation bug so I don't think we should add special logic for D3D12 in the CTS. We'll just do suppressions.

Reply:
Done.